lib/modules/dosomething/dosomething_global/dosomething_global.module

```diff
@@ -221,4 +221,37 @@ function dosomething_global_convert_country_to_language($country) {
     }
   }
   return 'en';
+
+/*
+ * Get the appropriate language for the application.
+ *
+ * @param $account
+ *   The user's account
+ * @param  $node
+ *   Optional node.
+ *
+ */
+function dosomething_global_get_language($account, stdClass $node = NULL) {
+  if ($node) {
+    // If there is a published translation for the node in the user's language, use it.
+    if ($node->translations->data['$account->language']['status']) {
+      $language = $account->language;
+    }
+    // If there's no published translation in their language try 'en-global'.
+    elseif ($node->translations->data['en-global']['status']) {
+      $language = 'en-global';
+    }
+    // Default to node language if there's no translation in their language and no 'en-global'.
+    else {
+      $language = $node->language;
+    }
+  }
+  elseif ($language = $account->language) {
+    return $language;
```

Comment:
This is an assignment, not a comparison. `if ($language = $account->language)` will always be `true`.

If the intended logic is "if the current language is equal to the account language, then just return the language here and skip the last few lines," that makes sense to me.